File: spec/azk/cli/commands/agent_spec.lua
local cli    = require('azk.cli')
local shell  = require('azk.cli.shell')
local helper = require('spec.spec_helper')
local io     = require('io')

local command = require('azk.cli.commands.agent')

describe("Azk #cli #command #agent", function()
  it("should execute a remove command in agent", function()
    local output = shell.capture_io(function()
      command.run("exec", "uname")
    end)

    assert.is.equal("Linux\n", output.stdout)
  end)

  it("should return a remote exit code", function()
    local result = command.run("exec", "/bin/bash", "-c", "exit 1")
    assert.is.equal(1, result)
  end)
end)

File: spec/cmds/agent_spec.js
import h from 'spec/spec_helper';
import { Cli } from 'azk/cli';

describe('Azk cli agent controller', function() {
  var outputs = [];
  var ui      = h.mockUI(beforeEach, outputs);

  var cli_options = {};
  var cli = new Cli(cli_options)
    .route('/agent');

  var doc_opts    = { exit: false };
  var run_options = { ui: ui };

  it("should run a agent status command", function() {
    doc_opts.argv = ['agent', 'status'];
    var options = cli.router.cleanArgs(cli.docopt(doc_opts));
    return cli.run(doc_opts, run_options).then((code) => {
      h.expect(code).to.equal(0);
      h.expect(options).to.have.property('agent', true);
      h.expect(options).to.have.property('status', true);
      h.expect(outputs[0]).to.match(RegExp(h.escapeRegExp('Agent is running...')));
    });
  });
});
